chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -221,39 +213,6 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
             .ok_or(PropertyQueryError::PrevBlockIndexNotFound(*prev_block_id))
     }
 
-    pub fn get_mainchain_tx_index(
-        &self,
-        tx_id: &OutPointSourceId,
-    ) -> Result<Option<TxMainChainIndex>, PropertyQueryError> {
-        log::trace!("Loading transaction index of id: {:?}", tx_id);
-        self.db_tx.get_mainchain_tx_index(tx_id).map_err(PropertyQueryError::from)
-    }
-
-    pub fn get_transaction_in_block(
-        &self,
-        id: Id<Transaction>,
-    ) -> Result<Option<SignedTransaction>, PropertyQueryError> {
-        log::trace!("Loading whether tx index is enabled: {}", id);
-        let is_tx_index_enabled = self.get_is_transaction_index_enabled()?;
-        if !is_tx_index_enabled {
-            return Err(PropertyQueryError::TransactionIndexDisabled);
-        }
-        log::trace!("Loading transaction index with id: {}", id);
-        let tx_index = self.db_tx.get_mainchain_tx_index(&OutPointSourceId::Transaction(id))?;
-        let tx_index = match tx_index {
-            Some(tx_index) => tx_index,
-            None => return Ok(None),
-        };
-        log::trace!("Loading transaction with id: {}", id);
-        let position = match tx_index.position() {
-            common::chain::SpendablePosition::Transaction(pos) => pos,
-            common::chain::SpendablePosition::BlockReward(_) => {
-                panic!("In get_transaction(), a tx id led to a block reward")
-            }
-        };
-        Ok(self.db_tx.get_mainchain_tx_by_position(position)?)
-    }
```

Comment:
I'm not sure if this is going to be an issue or not, but here `get_transaction_in_block()` returns `SignedTransaction` whereas the API Server currently returns a simplified transaction:

```
Ok(Json(json!({
    "block_id": block_id.map_or("".to_string(), |b| b.to_hash().encode_hex::<String>()),
    "version_byte": transaction.version_byte(),
    "is_replaceable": transaction.is_replaceable(),
    "flags": transaction.flags(),
    "inputs": transaction.inputs(),
    "outputs": transaction.outputs(),
    })))
```

Reply:
I'm not sure I understand how that can be a problem. The API server doesn't use anything internal from ChainstateRef. Also the API server analyzes individual blocks separately, not using the anything exposed from the node other than individual blocks.

Reply:
No problem, this was more an FYI because you mentioned today to flag anything that you think the API Server can't replace the transaction index with. All good.

Reply:
You can still including the signatures in the API server's analysis. That's all up to how it's implemented.